akka-actor/src/main/scala/akka/pattern/CircuitBreaker.scala

```diff
@@ -78,7 +77,14 @@ object CircuitBreaker {
  * @param resetTimeout [[scala.concurrent.duration.FiniteDuration]] of time after which to attempt to close the circuit
  * @param executor [[scala.concurrent.ExecutionContext]] used for execution of state transition listeners
  */
-class CircuitBreaker(scheduler: Scheduler, maxFailures: Int, callTimeout: FiniteDuration, resetTimeout: FiniteDuration)(implicit executor: ExecutionContext) extends AbstractCircuitBreaker {
+class CircuitBreaker(
+  scheduler:    Scheduler,
+  maxFailures:  Int,
+  callTimeout:  FiniteDuration,
+  resetTimeout: FiniteDuration,
+  factor:       Double         = 1.0)(implicit executor: ExecutionContext) extends AbstractCircuitBreaker {
```

Comment:
Must also add the old constructor for compatibility. We also have the `apply` and `create` methods in the companion. Perhaps we could instead add

```scala
def withExponentialBackoff(maxResetTimeout: FiniteDuration): CircuitBreaker
```

to create a new instance from an existing.

I don't think user need to care about the factor, we can just use `2.0`